Products/Zuul/catalog/model_catalog_tool.py

```diff
@@ -270,6 +271,28 @@ def getValue(obj):
         return sorted((unbrain(brain) for brain in queryResults),
                       key=getValue, reverse=reverse, cmp=natural_compare)
 
+    def get_all_docs(self):
+        """
+        Get all results from catalog by using solr deep pagination.
+        return generator.
+        """
+        cursor = INITIAL_CURSOR_MARK
+        limit = DEFAULT_SEARCH_LIMIT
+        order_by = INDEX_UNIQUE_FIELD
+        filterPermissions = False
+        query, _ = self._build_query(filterPermissions=filterPermissions)
+        need_result = True
+        while need_result:
+            search_params = SearchParams(query, limit=limit, order_by=order_by, next_cursor=cursor)
+            catalog_results = self.model_catalog_client.cursor_search(search_params, self.context)
+            if cursor == catalog_results.next_cursor:
+                need_result = False
+                break
+            cursor = catalog_results.next_cursor
+            for result in catalog_results.results:
+                yield result
```

Comment:
This looks _really_ similar to the `_wrap_results_gen` method in modelindex's `SolrSearcher` class.  Perhaps `SolrSearcher` should have the generator method instead.

I do not think the cursors should be treated identically to pagination.  Conceptually, cursors are very much like iterators and generators.  I think all the implementation detail of setting `cursorMark` and copying the cursor mark from the response into the next query should be kept in the `modelindex` package.  As far as `ModelCatalogTool` is concerned, it's receiving a iterator (or generator) that it has to iterate over to produce the brains.